system/src/system_network_manager_api.cpp

```diff
@@ -438,7 +438,7 @@ bool network_clear_credentials(network_handle_t network, uint32_t, NetworkCreden
     } else {
         if_t iface;
         if (!if_get_by_index(network, &iface)) {
-            return NetworkManager::instance()->clearConfiguration(iface);
+            return NetworkManager::instance()->clearConfiguration(iface) == 0;
```

Comment:
Looks good, similar to line 437.  Curious what the error was from, possibly SYSTEM_ERROR_INVALID_ARGUMENT due to the WiFi interface not being turned on before calling WiFi.clearCredentials() ?  Docs don't say it's required, but likely is.

Reply:
With a return type of bool there is not much other than success or fail

Reply:
I agree, I'm just wondering why this failed to return true in the first place.  Normally `clearConfiguration()` would return SYSTEM_ERROR_NONE (0), but the other possibility is SYSTEM_ERROR_INVALID_ARGUMENT which is a negative error code that would still cause this to return `false` with the new `== 0` if that is the case.  So what is causing that... possibly WiFi radio not powered on?